### ISSUE TITLE
Cannot overwrite DefaultServiceName when using OpenTracer

### ISSUE THREAD

**Issue by pgrm** [DataDog/dd-trace-dotnet#182]
**Describe the bug**

The OpenTracingSpan has functionality to overwrite the resource name and span type, but overwriting the service name is missing:

https://github.com/DataDog/dd-trace-csharp/blob/5e7656e/src/Datadog.Trace.OpenTracing/OpenTracingSpan.cs#L116-L139

**Expected behavior**

Calling `scope.Span.SetTag(DatadogTags.ServiceName, "a.new.service.name");` should overwrite the default service name defined in `OpenTracingTracerFactory.CreateTracer(new Uri(_configuration["datadog:endpointUri"]), "defult.service.name");`

**Additional context**

According to the unit tests defined here: https://github.com/DataDog/dd-trace-csharp/blob/0107d72a650c6536412b8db7b3393226418cfba3/src/Datadog.Trace.Tests/SpanBuilderTests.cs#L117-L142 this behavior should work in the normal datadog tracing span, but seemingly was forgotten in the OpenTracingSpan

I can't use datadog tracer, because I use opentracing for ef core and those 2 libraries don't seem to mix well as of now.

**Comment by willyyang** [comment]
Same if you call 
```
   var x = Tracer.Create(uri, "testme");
   var span = Tracer.Instance.StartSpan("csharpSpan");

   Log.Debug(span.ServiceName); <-- returns server
```

**Comment by lucaspimentel** [comment]
Thanks for reporting this! It will be fixed in the next release.

**Comment by lucaspimentel** [comment]
@willyyang: Note that `Tracer.Instance.StartSpan("csharpSpan")` sets the _operation_ name, not the service name. You want:
```
var span = Tracer.Instance.StartSpan("operation", serviceName: "service");
```

See: https://github.com/DataDog/dd-trace-dotnet/blob/cbf2af4c705f9517118a0047798da796977eb5e6/src/Datadog.Trace/Tracer.cs#L143

Let's limit this issue to using `OpenTracing.ISpan.SetTag()` to change the service name after a span is created. If you have further problems with `Tracer.Instance.StartSpan()`, please open a new issue.

**Comment by lucaspimentel** [comment]
@pgrm: We just released v0.6.0. It allows you to change the service name with `OpenTracing.ISpan.SetTag()`, so I'm closing this issue. Thanks again!
